### PR TITLE
Disambiguate targets

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: tools/generator/src/Generator+DistinguishTargets.swift [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        return [
+            platform.arch,
+            platform.minimumOsVersion,
+            platform.environment ?? "Device"
+        ].joined(separator: "-")
+    }
+}
+
+private extension String {
+    /// Computes an MD5 hash string for this `String`.
```
Comment (brentleyjones) [inline tools/generator/src/Generator+DistinguishTargets.swift:250]: I recently read that md5 is slower than sha1 on modern cpus, so I'll adjust this.

File: tools/generator/src/Generator+DistinguishTargets.swift [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+/// A `struct` containing values for a target that need to be unique in the eyes
+/// of Xcode, as well as the target itself.
+///
+/// Xcode requires that certain properties of targets are unique, such as name,
+/// the `TARGET_NAME` and `PRODUCT_MODULE_NAME` build settings, etc. This class
+/// collects information on all of the targets and calculates unique values for
+/// each one. If the values are user facing (i.e. the target name), then they
+/// are formatted in a readable way.
+struct DistinguishedTarget: Equatable {
```
Comment (brentleyjones) [inline tools/generator/src/Generator+DistinguishTargets.swift:12]: I had a hard time coming up with a name for this. I think `DistinguishedTarget` is _fine_, but should it be something else like `DisambiguatedTarget`?

**Comment by brentleyjones** [comment]
Current dependencies on/for this PR:
* main
  * **PR #22** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/22" target="_blank">Graphite</a> 
    * **PR #23** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/23" target="_blank">Graphite</a> 
      * **PR #24** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/24" target="_blank">Graphite</a> 
        * **PR #25** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/25" target="_blank">Graphite</a>  (this PR)
          * **PR #26** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/26" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: tools/generator/src/Generator+DistinguishTargets.swift (hunk 2 above)
Comment (tylerwilliams) [reply tools/generator/src/Generator+DistinguishTargets.swift:12]: Yeah both are kinda wordy. Disambiguated is at closer to what you I think this represents.

Naming is hard!

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**